.github/workflows/multi-arch-build.yaml

```diff
@@ -0,0 +1,119 @@
+    strategy:
+      matrix:
+        # where the Dockerfiles for builds are located at contrib/buildahimage directory
+        source:
+        - upstream
+        - stable
+    runs-on: ubuntu-latest
+    # internal registry for temporary build before push
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+          install: true
+
+      - name: Build and push local Buildah
+        uses: docker/build-push-action@v2
+        with:
+          context: contrib/buildahimage/${{ matrix.source }}
+          file: ./contrib/buildahimage/${{ matrix.source }}/Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          push: true
+          tags: |
+            localhost:5000/buildah/${{ matrix.source }}
+
+      # Simple verification that container works + get version for public image tag
+      - name: amd64 sniff test
+        id: run
+        env:
+          MATRIX_VERSION: ${{ matrix.source }}
+        run: |
+          # image tag is master for upstream source, actual version of buildah binary in the container for stable source
+          VERSION=v"$(docker run localhost:5000/buildah/${MATRIX_VERSION} buildah version | head -n 1 | awk '{print $2}')"
+          if [ "${MATRIX_VERSION}" == "upstream" ]; then
+            VERSION=master
+          fi
+          echo ::set-output name=version::$VERSION
+          echo ::set-output name=created::$(date -u --iso-8601=seconds)
+
+      # Push to buildah Quay repo stable and upstream Buildah (from matrix.source)
+      - name: Login to buildah Quay registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.BUILDAH_QUAY_REGISTRY}}
+          username: ${{ secrets.BUILDAH_QUAY_USERNAME }}
+          password: ${{ secrets.BUILDAH_QUAY_PASSWORD }}
+
+      - name: Push images to buildah Quay
+        uses: docker/build-push-action@v2
+        with:
+          cache-from: type=registry,ref=localhost:5000/buildah/${{ matrix.source }}
+          cache-to: type=inline
+          context: contrib/buildahimage/${{ matrix.source }}
+          file: ./contrib/buildahimage/${{ matrix.source }}/Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          push: true
+          tags: |
+            ${{ env.BUILDAH_QUAY_REGISTRY}}/${{ matrix.source }}:${{ steps.run.outputs.version }}
+          labels: |
+            org.opencontainers.image.source=${{ github.event.repository.html_url }}
+            org.opencontainers.image.revision=${{ github.sha }}
+            org.opencontainers.image.created=${{ steps.run.outputs.created }}
+
+      # Push to containers Quay repo only stable Buildah
+      - name: Login to containers Quay registry
```

Comment:
***RANT**: Oof, it would be nice to be able to use YAML anchor/aliases here (IIRC, *still* not supported by github-actions).

Reply:
no support for anchors so far is also my understanding. And composite actions only for shell steps - https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/

Reply:
Yeah, it's one of the most annoying things about github-actions compared to gitlab CI/CD.  Also, not being able to reuse one common set of steps into multiple workflows. In fact, I'm contemplating it might be worth adding a comment about this.  Just to save somebody an attempt toward deduplicating the config (since that's impossible at the moment).  I think these are different enough it's probably fine as-is.

Reply:
still added comment about it